Add WithBody() function -- Issue #11

benpate opened the pull request:
This checkin adds a “WithBody” function and enables Wrecker to post
JSON-encoded documents directly to a REST web service.

In the process, we’re simplifying the signature of the sendRequest()
function by simply passing in the entire WreckerRequest object.

benpate committed:
Add WithBody() function (Issue #11)

This checkin adds a “WithBody” function and enables Wrecker to post
JSON-encoded documents directly to a REST web service.

In the process, we’re simplifying the signature of the sendRequest()
function by simply passing in the entire WreckerRequest object.

benpate commented:
Same comment as before.  I'm still having trouble running the automated tests.  I'll re-import all of your changes, and have a go at making test cases for these updates.

BrandonRomano commented:
Sounds good -- this is looking pretty good so far.  I'll give this a look once the test cases are up!

benpate committed:
[Issue #11] Test Cases

These test cases demonstrate wrecker working with JSON data in the
request body

benpate commented:
Just posted test cases.  Does this work for you?

BrandonRomano commented:
Hey Ben -- I'll hopefully be able to check these out sometime tomorrow!

BrandonRomano commented:
@benpate I'm thinking of adding you as a contributor to this project (so you can directly work in this repo... will generally speed things up).

Shoot me an email at BrandonRRomano@gmail.com and we can sort out a workflow between us, if you'd like!

BrandonRomano commented:
Hey @benpate, will be giving this a full look over tonight.

I'm going to also push your changes up to a branch in this repo (as opposed to your fork), and we'll PR that as opposed to this so we can get started with our new flow

BrandonRomano commented:
Closing this PR in favor of #14 (which is an in-project rebased version of this PR)
